### ISSUE TITLE
[Request]: Move @username inline with name

### ISSUE THREAD

**Issue by bertob** [GeopJr/Tuba#159]
### Describe the request

The @username is currently below the name, which makes posts taller than they need to be, creates unnecessary whitespace on the right, and doesn't look great visually. I'd move it up next to the name, like most other clients do.

![image](https://user-images.githubusercontent.com/1908896/230046804-a4c09b7d-7458-410f-ab8a-3fcab58bff7a.png)

Elk:

![image](https://user-images.githubusercontent.com/1908896/230045158-2086ef6f-5fe7-4f75-9c5c-2e4c3b32497e.png)

Phanpy:

![image](https://user-images.githubusercontent.com/1908896/230046470-10e9b921-f5a2-427b-a87f-d0fd17eb8c7b.png)

Ivory:

![image](https://user-images.githubusercontent.com/1908896/230047510-dd9fed99-3e99-45c0-8a0c-062ba55d79d0.png)

### Implementation Details

- [ ] This should be an option in settings.
- [ ] This should be only available to some fediverse backends. (Include which ones on the above field).
- [X] This is client-only (and shouldn't sync with the instance).
- [X] This follows the [GNOME HIG](https://developer.gnome.org/hig/).

**Comment by GeopJr** [comment]
Since this will change the layout of the status, it might be better to combine it with #62 as to avoid future revisits:

additional post actions (like Edit or Delete) are in the context menu, it might be better to have them as a button since they are relatively hidden.

- Elk places the button next to the date, but also doesn't display post indicators (visibility, pinned, edited icons)
- Mastodon places it next to the post actions, but doesn't display a dedicated bookmark button. - Mastodon mobile app places it at the top right but doesn't display indicators and the date is next to the handle

**Comment by bertob** [comment]
I think both the top right and bottom right in the action bar would work layout-wise, semantically I think I have a slight preference for top right.

**Comment by GeopJr** [comment]
I went with top right!

The main issue with the inline handle is wrapping and spacing. For example the following post:

![Screenshot from 2023-04-10 12-45-37](https://user-images.githubusercontent.com/18014039/230885538-f47fe386-47ba-44e0-9a52-4139a5234c65.png)

If the content moves higher, it doesn't leave much space for the button. If the name + handle + indicators + button also move higher, the button will reach the container limits

![Screenshot from 2023-04-10 13-05-59](https://user-images.githubusercontent.com/18014039/230886105-c52b6cc1-61b0-4cff-a76d-8f601eba253f.png)

As for wrapping, elk ellipsizes both the name and the title (in wide mode / when the handle is inline):
![image](https://user-images.githubusercontent.com/18014039/230886388-f5362183-8146-47ba-b297-d8116092cd8d.png)

but in narrow mode it puts it places it below the name:

![image](https://user-images.githubusercontent.com/18014039/230886597-b1aa4b4c-8eba-4fcb-a243-c6731cc169f6.png)

I tried multiple different routes:

Having only the handle ellipsize leads to situations such as this

![Screenshot from 2023-04-10 13-10-45](https://user-images.githubusercontent.com/18014039/230886718-51d605dd-1aa4-4bbf-98f5-dc37e0f0d3d0.png)

Having both ellipsize gets (imo) way too much info omitted (especially in narrow mode) 

![Screenshot from 2023-04-10 13-11-57](https://user-images.githubusercontent.com/18014039/230886864-3d5e2817-e82e-47bc-8834-76e2f66f3523.png)

Using flowboxes works but they require additional space below them (plus it might be inconsistent as a status from someone with a short name and handle will be inline but right below it a status from someone with a long one, won't be) 

![Screenshot from 2023-04-10 13-27-03](https://user-images.githubusercontent.com/18014039/230887052-0f7024e9-a2bb-480c-9009-8b03ee9503b2.png)
![Screenshot from 2023-04-10 13-26-52](https://user-images.githubusercontent.com/18014039/230887054-10333006-60ba-4519-8980-9797505f9897.png)

**Comment by bertob** [comment]
> If the content moves higher, it doesn't leave much space for the button. If the name + handle + indicators + button also move higher, the button will reach the container limits

I'd move it up just enough so it doesn't touch the button (maybe 2px margin below the button?). Elk seems to do something similar:


Is there a branch with the current state?

**Comment by GeopJr** [comment]
Yep: https://github.com/GeopJr/Tuba/tree/feat/status/title-redesign (it's uses flowboxes solution)

**Comment by bertob** [comment]
Hmmm yeah, the spacing below the header would need to be collapsed when the username is inline, otherwise there's this weird whitespace below.

![image](https://user-images.githubusercontent.com/1908896/230994856-0c9fe0b5-ed4f-4f4e-bd50-f01fa6cfa44c.png)

Also, looking at Elk in comparison I realized that one reason why the vertical spacing looks better there is because they're reducing the height of newlines. We should probably do the same, though I'd maybe go a little bit taller than what they have.

![image](https://user-images.githubusercontent.com/1908896/230994784-e36abe55-3373-4a50-ae29-d631a6ff46b1.png)

**Comment by GeopJr** [comment]
The header should now look as expected, but it required the status to migrate from grid to nested boxes and some margins might have been rendered useless. I think I caught everything but if you see anything that doesn't match how it looked before, let me know!

One unfortunate side-effect of that change is that expanding a post is not possible anymore:

before:

![image](https://user-images.githubusercontent.com/18014039/231630356-44cf3649-7cef-4101-96c6-f7353ff13ca9.png)

after:

![image](https://user-images.githubusercontent.com/18014039/231630429-accf22fc-7cf8-4a42-b67b-774c2ffcb229.png)

If that's a deal-breaker I could try some workarounds.


About the username, it can move higher up if needed:
![image](https://user-images.githubusercontent.com/18014039/231630891-99676fbb-4866-4486-a317-bb9e2a7db84c.png)

**Comment by bertob** [comment]
Where was expanding posts used previously? Doesn't seem like a dealbreaker to me, but maybe I'm missing something.

As for moving the user name up, I'd actually go the other way and move it down a bit more, so the horizontal spacing between the avatar and the name is more similar to the vertical spacing between name and content. Alternatively we could also just increase the horizontal spacing left of the text column, maybe worth trying a few alternatives.

**Comment by GeopJr** [comment]
The post that opens a thread is expanded:

before:
![Screenshot from 2023-04-17 00-01-58](https://user-images.githubusercontent.com/18014039/232342121-627bb4bd-b11b-4d02-8ed6-c348345325e6.png)

after:
![Screenshot from 2023-04-17 00-01-23](https://user-images.githubusercontent.com/18014039/232342133-6e0eac1f-6a95-43ce-9941-ef08f4a89fc7.png)


About the name:

Decreasing the spacing between the name and content:
![Screenshot from 2023-04-16 23-55-40](https://user-images.githubusercontent.com/18014039/232342326-ae5b4e5b-0eea-4740-a781-3891f5190eca.png)

Increasing the spacing between the avatar and the rest:
![Screenshot from 2023-04-16 23-58-18](https://user-images.githubusercontent.com/18014039/232342353-96ca05ee-814b-4363-89b1-796bac703e60.png)

Increasing the spacing between the avatar and just the name:
![image](https://user-images.githubusercontent.com/18014039/232342448-4e5c08bb-4aa9-4e2c-8743-0e68ba5631a3.png)


I'm leaning towards increasing the spacing between the avatar and the name + content (the second one)

**Comment by bertob** [comment]
Interesting, I'd never really paid attention to this, but it looks like most other apps do this expanding thing too. There's a few other differences in how other apps do it though, namely
- Expanded posts have larger body text
- Date/time/other metadata is moved to the bottom to make more space for long names/usernames
- Longer, more detailed format for date/time

I think it'd make sense to have something like that since everyone else does it, but then I'd also make it a bit more distinct from normal posts, with the bigger font size, different metadata layout, etc.

![image](https://user-images.githubusercontent.com/1908896/232474070-ea4cd24e-906e-440a-8d69-5602790bf162.png)

![image](https://user-images.githubusercontent.com/1908896/232474701-297f8ad4-6ade-4845-ac5f-8f6dd8fa210b.png)

> I'm leaning towards increasing the spacing between the avatar and the name + content (the second one)

Agreed!

**Comment by bertob** [comment]
Looks great, except for a few minor things:
- The name and username are a little bit too far apart vertically, I'd decrease the vertical spacing there by like 3px (but keep them vertically centered on the avatar)
- I'd add 6px of spacing above the date/time metadata line at the bottom
- I'd use a more readable format for the long date, I'd try something like "February 5, 2023 · 18:20"

**Comment by GeopJr** [comment]
![image](https://user-images.githubusercontent.com/18014039/232639499-2e6c2f27-d2e8-4787-8cd0-1f5acf0abc95.png)

Unsure about the post visibility indicator, there's also the edited indicator
![image](https://user-images.githubusercontent.com/18014039/232639878-aff02f1c-936d-44dc-8127-713e7836416e.png)
![image](https://user-images.githubusercontent.com/18014039/232640107-63b24d49-7262-426c-8c32-07abd86e9452.png)

also somewhat better spacing:
![image](https://user-images.githubusercontent.com/18014039/232640188-2f254b68-c542-4689-b1b8-976315f8edbf.png)

FWIW, the indicators already have tooltips:
![image](https://user-images.githubusercontent.com/18014039/232640248-f30e6e76-7983-4d8c-be23-a1cce700cb7a.png)

**Comment by bertob** [comment]
Hmm yeah, maybe it was better with just an icon, indeed. I think the real issue is maybe that the icon looks so large next to the non-bold text. Maybe we just need smaller variants of these icons, I already noticed this in the context of the icon in the top right. Is there a list of all the icons are we using in this context, so I can draw smaller ones?

Other than that this looks good to me, though I'd add middle dot separators between all the fields, not just between date and time.

**Comment by GeopJr** [comment]
I don't think you need to re-draw them, we can decrease the pixel-size (either for both normal and expanded mode or just expanded):

16 / normal size
![Screenshot from 2023-04-18 16-56-37](https://user-images.githubusercontent.com/18014039/232802384-56186396-db32-45bb-85f2-9602db7e599f.png)

14
![Screenshot from 2023-04-18 16-56-30](https://user-images.githubusercontent.com/18014039/232802440-8e08c215-8cc0-430f-b8dd-4afc9ccedd25.png)

![image](https://user-images.githubusercontent.com/18014039/232803636-8ec10bb5-73f4-4907-9c45-a8552ffd9cb2.png)


otherwise the icons are: `tuba-globe-symbolic`, `tuba-mail-symbolic`, `tuba-padlock2-open-symbolic`, `tuba-padlock2-symbolic` and `document-edit-symbolic`

(the ones with the tuba- prefix are located in `data/icons/scalable/actions/`)

**Comment by bertob** [comment]
> I don't think you need to re-draw them, we can decrease the pixel-size (either for both normal and expanded mode or just expanded):

Well, the problem is that then they're off the pixel grid and look blurry. Maybe for now try and use them at the smaller size, and we can substitute them for hinted ones later.

Also, I noticed that the client name doesn't seem to be dimmed, so it looks odd when it's not a link:

![image](https://user-images.githubusercontent.com/1908896/232808553-2875dcdb-b692-4dc3-9cd7-b8fec7db8603.png)


Other than that and the things mentioned above, this looks good to me now.

**Comment by GeopJr** [comment]
Everything should be done, I still don't know how to feel about the visibility text (especially when the post is also edited):
![image](https://user-images.githubusercontent.com/18014039/232812862-9e75ea3f-f5de-4288-94da-093d78f6a549.png)
![image](https://user-images.githubusercontent.com/18014039/232813146-72eda254-244d-42f2-b3af-aa1dca9b3ac6.png)

I'll clean the PR and merge it later, until then there's still time for any final notes if needed!

**Comment by bertob** [comment]
> Everything should be done, I still don't know how to feel about the visibility text (especially when the post is also edited):

As I said, I think at the smaller size icon-only wold be fine here.

**Comment by bertob** [comment]
Opened a devkit issue to track the small icon thing https://gitlab.gnome.org/Teams/Design/icon-development-kit/-/issues/80
